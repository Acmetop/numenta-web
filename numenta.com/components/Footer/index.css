/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

.footbar {
  composes: w-100 from 'tachyons';
}

.footer {
  composes: br--bottom from 'tachyons';
  composes: br2 from 'tachyons';
  composes: center from 'tachyons';
  composes: f6 from 'tachyons';
  composes: mw-100 from 'tachyons';
  composes: mw7-m from 'tachyons';
  composes: mw8-l from 'tachyons';
  composes: pv4 from 'tachyons';
  composes: silver from 'tachyons';
  composes: tc from 'tachyons';
}

.mark {
  composes: dib from 'tachyons';
  composes: h1 from 'tachyons';
  composes: mh3 from 'tachyons';
  composes: relative from 'tachyons';
  composes: w1 from 'tachyons';
  top: 0.4rem;
}
