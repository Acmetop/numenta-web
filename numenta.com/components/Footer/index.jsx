// Numenta company website source code. Copyright © 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import moment from 'moment'
import React from 'react'

import {getRepo} from '../../utils/shared'

import LogoMark from '../LogoMark'
import Spacer from '../Spacer'
import TextLink from '../TextLink'

import styles from './index.css'


const Footer = (props, {config}) => {
  const {links} = config
  const repo = getRepo()
  const year = moment().year()

  return (
    <div className={styles.footbar}>
      <footer className={styles.footer}>
        <span>
          © {year} {' '}
          <TextLink to={links.in.home}>Numenta</TextLink>
        </span>
        <span className={styles.mark}>
          <LogoMark />
        </span>
        <span>
          <TextLink to={links.in.terms}>Terms</TextLink>
          <Spacer />
          <TextLink to={links.in.privacy}>Privacy</TextLink>
          <Spacer />
          <TextLink to={links.in.sitemap}>Sitemap</TextLink>
          <Spacer />
          <TextLink to={repo}>Source</TextLink>
        </span>
      </footer>
    </div>
  )
}

Footer.contextTypes = {
  config: React.PropTypes.object,
}

export default Footer
