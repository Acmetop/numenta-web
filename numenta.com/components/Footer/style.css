.footbar {
  composes: w-100 from 'tachyons';
}

.footer {
  composes: br--bottom from 'tachyons';
  composes: bg-white from 'tachyons';
  composes: br2 from 'tachyons';
  composes: center from 'tachyons';
  composes: f6 from 'tachyons';
  composes: mw-100 from 'tachyons';
  composes: mw7-m from 'tachyons';
  composes: mw8-l from 'tachyons';
  composes: pv3 from 'tachyons';
  composes: silver from 'tachyons';
  composes: tc from 'tachyons';
}

.mark {
  composes: dib from 'tachyons';
  composes: h1 from 'tachyons';
  composes: mh3 from 'tachyons';
  composes: relative from 'tachyons';
  composes: w1 from 'tachyons';
  top: 3px;
}
