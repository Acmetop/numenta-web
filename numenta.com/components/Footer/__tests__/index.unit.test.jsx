import React from 'react'
import renderer from 'react-test-renderer'
import stubContext from 'react-stub-context'

import context from '../../../__mocks__/reactContextMock'

import Footer from '../../Footer'


describe('<Footer /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const FooterStubbed = stubContext(Footer, context)
    const component = renderer.create(
      <FooterStubbed />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
