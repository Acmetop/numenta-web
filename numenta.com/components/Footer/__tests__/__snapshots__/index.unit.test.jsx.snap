exports[`Footer React component Renders correctly 1`] = `
<div
  className="footbar">
  <footer
    className="footer">
    <span>
      © 
      2016
       
       
      <a
        className="textlink"
        href={undefined}
        onClick={[Function bound handleClick]}
        style={Object {}}
        target={undefined}>
        Numenta
      </a>
    </span>
    <span
      className="mark">
      <img
        alt="Numenta Logo Mark"
        className="image"
        onClick={undefined}
        src="filename.ext"
        title="Numenta" />
    </span>
    <span>
      <a
        className="textlink"
        href={undefined}
        onClick={[Function bound handleClick]}
        style={Object {}}
        target={undefined}>
        Terms
      </a>
      <span
        className="spacer">
        •
      </span>
      <a
        className="textlink"
        href={undefined}
        onClick={[Function bound handleClick]}
        style={Object {}}
        target={undefined}>
        Privacy
      </a>
      <span
        className="spacer">
        •
      </span>
      <a
        className="textlink"
        href={undefined}
        onClick={[Function bound handleClick]}
        style={Object {}}
        target={undefined}>
        Sitemap
      </a>
      <span
        className="spacer">
        •
      </span>
      <a
        className="textlink"
        href="https://github.com/numenta/numenta-web"
        onClick={[Function instrumentOnClick]}
        target={undefined}>
        Source
      </a>
    </span>
  </footer>
</div>
`;
